index.coffee

```diff
@@ -159,11 +159,13 @@ class ProgressReporter
 				console.warn('Progress error:', err.message ? err)
 
 exports.DockerProgress = class DockerProgress
-	constructor: (dockerOpts) ->
+	constructor: (opts = {}) ->
 		if !(this instanceof DockerProgress)
-			return new DockerProgress(dockerOpts)
-
-		@docker = new Docker(dockerOpts)
+			return new DockerProgress(opts)
+		if opts.docker?
+			@docker = opts.docker
```

Comment:
With a passed in docker we have no idea what promise library it will be using - however we rely on bluebird sugar being available on the returned promises (which also applies to an opts object case as we now use the built-in dockerode promise funcs rather than the promisifyAll versions)